include/ESA.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef ESA_HPP
```

Comment:
I suggest renaming the files ESA.hpp and ESA.cpp to esa.hpp and esa.hpp in order to be consistent with the naming of all the other c++ source files